alphagov/pay-connector
# PP-5761 Modify MDC logging to add charge and account info

Register a common logging request filter for all requests to enrich MDC
data. If either the `chargeId` or `accountId` exist as path parameters then look up
those entities and add information to MDC to appear in future log lines.

This is in addition to the existing `LoggingRequestFilter` which just
logs the start and end of each request.

This commit repurposes the `ChargeIdLoggingMDCRequestFilter` by renaming it and
adding the ability to get account information. It also registers it for
every request rather than as a dynamic feature.

## How to test
Check out the CI logs and see the new keys, for example:
```
{"@timestamp":"2019-11-27T11:47:34.206Z","@version":"1","message":"Card refund request sent - charge_external_id=abc123, status=CAPTURED, amount=100, transaction_id=aGatewayTransactionId, account_id=42, operation_type=Refund, amount_available_refund=100, amount_requested_refund=100, provider=sandbox, provider_type=test, user_external_id=3b7b5f33-24ea-4405-88d2-0a1b13efb20c","logger_name":"uk.gov.pay.connector.refund.service.ChargeRefundService$$EnhancerByGuice$$f289fb35","thread_name":"dw-352 - POST /v1/api/accounts/42/charges/abc123/refunds","level":"INFO","level_value":20000,"gateway_account_id":"42","payment_external_id":"abc123","provider":"sandbox","gateway_account_type":"test","container":"connector"}
```
